modules/earthengine-layers/src/earth-engine-layer.js

```diff
@@ -108,78 +103,38 @@ export default class EarthEngineLayer extends CompositeLayer {
-        id: this._getLayerId(getTileUrl),
-        async getTileData({x, y, z}) {
-          const imageUrl = getTileUrl(x, y, z);
-          const image = await loadImageBitmap(imageUrl);
-          // const imageData = await getImageData(image);
-
-          const bounds = merc.bbox(x, y, z);
-
-          // const meshGrid = createMeshGrid(bounds, imageData);
-          // const terrain = getGrayScaleData(imageData);
-          // const tile = martini.createTile(terrain);
-          // const mesh = tile.getMesh(10);
-          // console.debug(`loaded tile ${x}/${y}/${z}`, bounds);
-
-          return {
-            id: `${z}/${x}/${y}`,
-            image,
-            bounds
-          };
-        },
-
-        renderSubLayers(props) {
-          // console.debug(props.data.meshGrid);
-
-          return (
-            props.data.image && [
-              /*
-            new SimpleMeshLayer({
-              ...props.data,
-              data: [{0, 0}]
-              mesh: {
-                header: {vertexCount: 0},
-                attributes: {
-                  POSITION: {},
-                  NORMAL: {},
-                  TEXCOORD_0: {value}
-                }
-              }
-            }),
-            new ColumnLayer({
-              id: `${props.id}-pt`,
-              data: props.data.meshGrid,
-              extruded: true,
-              radius: 30,
-              getPosition: d => [d[0], d[1]],
-              getRadius: 0.01,
-              getFillColor: d => [0, d[2][1], d[2][2]],
-              getElevation: d => d[2][0] * 10,
-              opacity: 1
-            }),
-            */
-              // new ScatterplotLayer({
-              //   id: `${props.id}-pt`,
-              //   data: props.data.meshGrid,
-              //   extruded: true,
-              //   radius: 30,
-              //   getPosition: d => [d[0], d[1]],
-              //   getRadius: 0.01,
-              //   getFillColor: d => [0, d[2][1], d[2][2]],
-              //   getElevation: d => d[2][0] * 10,
-              //   opacity: 1
-              // }),
-              new BitmapLayer({
-                ...props.data
-                // opacity: 0.2
-              })
-            ]
-          );
+      new TileLayer(
+        this.getSubLayerProps({
+          id: 'tiles'
+        }),
+        {
+          id: this._getLayerId(getTileUrl),
+          async getTileData({x, y, z}) {
+            const imageUrl = getTileUrl(x, y, z);
+            const image = await loadImageBitmap(imageUrl);
+            // const imageData = await getImageData(image);
+
+            return image;
+          },
+
+          renderSubLayers(props) {
+            const {data, tile} = props;
+            const {west, south, east, north} = tile.bbox;
+            const bounds = [west, south, east, north];
+
+            return (
+              data && [
+                new BitmapLayer(
+                  Object.assign(props, {
```

Comment:
There is a function in `CompositeLayer` for "controlled" forwarding of props. Recommend looking at other CompositeLayers and following this model if it makes sense.

That could be a second PR.

Reply:
Is that `this.getSubLayerProps`? I think I found it in another composite layer. I use it above for getting `TileLayer` props, and maybe it's ideal to use it again for the `BitmapLayer` props.